pkg/apis/networking/v1alpha1/ingress_types.go

```diff
@@ -109,6 +114,18 @@ type IngressSpec struct {
 	DeprecatedVisibility IngressVisibility `json:"visibility,omitempty"`
 }
 
+type HTTPOption string
+
+const (
+	// The knative ingress will be able to serve HTTP connections.
+	HTTPOptionEnabled HTTPOption = "Enabled"
+	// The knative ingress will reject HTTP traffic
+	HTTPOptionDisabled HTTPOption = "Disabled"
```

Comment:
@ZhiminXiang I think we talked about removing this and biasing towards removing the exposure from kingress gateways.  Part of this was motivated by not wanted to work out the interaction with HTTP01 TLS challenges IIRC.

Reply:
yes, we had the discussion in the design doc.

Here Victor and I thought we could still keep it in the IngressSpec but make Route controller not set "Disabled" option. With this way, the HTTP01 challenge will not be interacted within Serving. And it potentially makes it easy if in the future we want to support "Disabled".  WDYT?

Reply:
It's either part of conformance or not.  The sentiments I was talking about in the doc weren't specific to Route at all, but all clients of the networking API.